src/services/data-fetch/user-pages-data-fetch/team-page-data-fetch/TeamPageDataFetch.ts

```diff
@@ -1,3 +1,53 @@
-import { TeamPageMock } from '../../../../utils/mock-data/user-pages/team-page/TeamPage';
+import {
+    Member,
+    PublicCategoryWithTeamMembersDto,
+    PublicTeamMemberDto,
+    TeamItem,
+    TeamPageData
+} from '../../../../types/TeamPage';
+import default_team_member_photo from '../../../../assets/team_page_images/team_member_not_found_photo.svg';
+import { axiosInstance } from '../../../api/axios';
+import {API_ROUTES} from '../../../../const/api-routes-constants'
 
-export const teamPageDataFetch = async () => TeamPageMock;
+const isValidCategory = (category: PublicCategoryWithTeamMembersDto): boolean => {
+    return Boolean(
+        category?.categoryName?.trim() &&
+        Array.isArray(category.teamMembers) &&
+        category.teamMembers.length > 0
+    );
+};
+
+const isValidTeamMember = (member: PublicTeamMemberDto): boolean => {
+    return Boolean(member?.fullName?.trim());
```

Comment:
Just a suggestion (not nessesary to fix)
In order to transform value to boolean you can use [double NOT](double NOT) operator